src/domain/session/room/RoomViewModel.js

```diff
@@ -67,6 +67,29 @@ export class RoomViewModel extends ViewModel {
         this._clearUnreadAfterDelay();
     }
 
+    async _recreateComposerOnPowerLevelChange() {
+        const powerLevelObservable = await this._room.observePowerLevels();
+        let oldCanSendMessage = powerLevelObservable.get().canSendType("m.room.message");
```

Comment:
```suggestion
        const canSendMessage = () => powerLevelObservable.get().canSendType("m.room.message");
        let oldCanSendMessage = canSendMessage();
```
And reuse that below as well?